### PR TITLE
Delete Debtor Groups

### PR DESCRIPTION
**Pull request by lomamech**
@sfount  @jniles 

Delete Debtor Groups  …
- Adding service delete in DebtorGroupService
- Update interface for Debtor Groupe Management
- Update API for Debtor Group
- Adding test integration for Debtor Group
- Adding test E2E for deletion of Debtor Group`

### CODE REVIEW

File: client/src/i18n/fr.json [hunk 1]
```diff
@@ -188,7 +188,9 @@
     "DEBTOR"         : {
      "SUBSCRIBED"    : "Debiteurs souscris"
     },
+    "DELETE"         : "Supprimer un groupe débiteur",
     "EDIT"           : "Editer le groupe debiteur",
+    "FAILURE_DELETE" : "Echec de suppression du groupe créditeur",
```
Comment (jniles) [inline client/src/i18n/fr.json]: Je pense qu'il doit dire "du group debiteur"

File: client/src/partials/debtors/groups.list.html [hunk 2]
```diff
@@ -35,7 +35,8 @@
                 <h4 style="margin-top : 0px;"><span ng-if="debtorGroup.locked" class="fa fa-lock text-danger"></span> <b>{{debtorGroup.name}}</b></h4>
               </div>
               <div class="col-md-6 text-float-util">
-                <a data-method="update" ui-sref="debtorGroups.update({uuid : debtorGroup.uuid})" class=""><span class="fa fa-pencil-square-o" aria-hidden="true"></span> {{ "DEBTOR_GROUP.EDIT" | translate }}</a>
+                <a data-method="update" ui-sref="debtorGroups.update({uuid : debtorGroup.uuid})" class=""><span class="fa fa-pencil-square-o" aria-hidden="true"></span> {{ "DEBTOR_GROUP.EDIT" | translate }}</a><br>
```
Comment (jniles) [inline client/src/partials/debtors/groups.list.html]: These should use the translate directive.

[Documentation](https://angular-translate.github.io/docs/#/guide/05_using-translate-directive)

File: server/controllers/finance/debtors/groups/index.js [hunk 3]
```diff
@@ -295,3 +298,24 @@ function loadInvoices(params) {
...
+ * @description
+ * This method removes the debtor group from the system.
+ */
+function remove(req, res, next) {
+  const sql = 'DELETE FROM debtor_group WHERE uuid = ?;';
+  const uid = db.bid(req.params.uuid);
+  db.exec(sql, [uid])
+    .then(rows => {
+      if (!rows.affectedRows) {
+        throw new BadRequest(`Cannot delete the debtor group with id ${req.params.uuid}`, 'DEBTOR_GROUP.FAILURE_DELETE');
```
Comment (jniles) [inline server/controllers/finance/debtors/groups/index.js]: Apparently BadRequest isn't defined!

File: test/end-to-end/debtors/debtorgroups.spec.js [hunk 4]
```diff
@@ -38,6 +38,17 @@ describe('Debtor Groups Management', function () {
     expect(element.all(by.css('[data-group-entry]')).count()).to.eventually.equal(initialGroups + 1);
   });
 
+  it('Delete a debtor group', function () {
```
Comment (jniles) [inline test/end-to-end/debtors/debtorgroups.spec.js]: :+1:

File: test/integration/debtorGroups.js [hunk 5]
```diff
@@ -297,6 +297,19 @@ describe('(/debtor_groups) The debtor groups API', function () {
     .catch(helpers.handler);
   });
 
+  it('DELETE /debtor_groups/:uuid should delete an existing debtor group', function () {
```
Comment (jniles) [inline test/integration/debtorGroups.js]: :+1:

**Comment by lomamech** [comment]
image

All E2E tests pass but not with Travis 

@jniles

**Pull request merged**